e2e/node_test.go

```diff
@@ -37,7 +37,12 @@ func testNode() {
 				return err
 			}
 
-			if len(podlist.Items) != 3 {
+			count := 3
+			if isDaemonsetLvmdEnvSet() {
+				count = 1
+			}
+
+			if len(podlist.Items) != count {
```

Comment:
https://github.com/topolvm/topolvm/pull/265/files#r595008742
> I assume this test checks topolvm-node should be deployed on all nodes,
so check if pod number equals worker node number.

when using kind, I need to exclude control plane nodes because control plane nodes don't deploy non control plane pods to themselves.

```
$ kubectl get node --selector=node-role.kubernetes.io/control-plane!=''
NAME                  STATUS   ROLES    AGE     VERSION
topolvm-e2e-worker    Ready    <none>   5m50s   v1.20.2
topolvm-e2e-worker2   Ready    <none>   5m50s   v1.20.2
topolvm-e2e-worker3   Ready    <none>   5m50s   v1.20.2
```

But, on the other hand, when using minikube(bare metal), if I exclude control plane nodes, I can't find the worker node because the worker node is also a control plane node.

```
$ kubectl get node --selector=node-role.kubernetes.io/control-plane!=''
No resources found
```

So I think it is difficult to compare the counts of pods and nodes dynamically.

Reply:
Thank you for your explanation.
I got it.